src/app/modules/submission/email-submission/__tests__/email-submission.routes.spec.ts

```diff
@@ -0,0 +1,368 @@
+              signature: Joi.string().allow(''),
+            })
+            .xor('answer', 'answerArray') // only answer or answerArray can be present at once
+            .with('filename', 'content'), // if filename is present, content must be present
+        )
+        .required(),
+      isPreview: Joi.boolean().required(),
+    }),
+  }),
+  EmailSubmissionsMiddleware.validateEmailSubmission,
+  MyInfoController.verifyMyInfoVals as RequestHandler,
+  (SubmissionsController.injectAutoReplyInfo as unknown) as RequestHandler,
+  SpcpController.appendVerifiedSPCPResponses as RequestHandler,
+  EmailSubmissionsMiddleware.prepareEmailSubmission as RequestHandler,
+  EmailSubmissionsMiddleware.saveMetadataToDb,
+  EmailSubmissionsMiddleware.sendAdminEmail,
+  SubmissionsController.sendAutoReply,
+)
+
+const EmailSubmissionsApp = setupApp('/', EmailSubmissionsRouter)
+
+describe('email-submission.routes', () => {
+  let request: Session
+
+  beforeAll(async () => await dbHandler.connect())
+  beforeEach(() => {
+    request = session(EmailSubmissionsApp)
+  })
+  afterEach(async () => {
+    await dbHandler.clearDatabase()
+    jest.restoreAllMocks()
+  })
+  afterAll(async () => await dbHandler.closeDatabase())
+
+  describe('SPCP authentication', () => {
+    describe('SingPass', () => {
+      it('should return 200 when submission is valid', async () => {
+        mockSpClient.verifyJWT.mockImplementationOnce((_jwt, cb) =>
+          cb(null, {
+            userName: 'S1234567A',
+          }),
+        )
+        const { form } = await dbHandler.insertEmailForm()
+        // authType cannot be saved if form is published, so need to save twice
+        form.esrvcId = 'mockEsrvcId'
+        form.authType = AuthType.SP
+        form.hasCaptcha = false
+        await form.save()
+        form.status = Status.Public
+        await form.save()
+
+        const response = await request
+          .post(`${SUBMISSIONS_ENDPT_BASE}/${form._id}`)
+          .field('body', JSON.stringify(MOCK_SUBMISSION_BODY))
+          .query({ captchaResponse: 'null' })
+          .set('Cookie', ['jwtSp=mockJwt'])
+
+        expect(response.status).toBe(200)
+        expect(response.body).toEqual({
+          message: 'Form submission successful.',
+          submissionId: expect.any(String),
+        })
+      })
+
+      it('should return 401 when submission does not have JWT', async () => {
+        const { form } = await dbHandler.insertEmailForm()
+        // authType cannot be saved if form is published, so need to save twice
+        form.esrvcId = 'mockEsrvcId'
+        form.authType = AuthType.SP
+        form.hasCaptcha = false
+        await form.save()
+        form.status = Status.Public
+        await form.save()
+
+        const response = await request
+          .post(`${SUBMISSIONS_ENDPT_BASE}/${form._id}`)
+          .field('body', JSON.stringify(MOCK_SUBMISSION_BODY))
+          .query({ captchaResponse: 'null' })
```

Comment:
nit: Add comment to note the lack of jwt cookie? Easier to pinpoint the expected data in this test